internal/controllers/machineset/machineset_controller.go

```diff
@@ -1604,3 +1603,19 @@ func (r *Reconciler) reconcileExternalTemplateReference(ctx context.Context, clu
 
 	return false, patchHelper.Patch(ctx, obj)
 }
+
+// Returns the machines to be remediated in the following order
+//   - Machines with RemediateMachineAnnotation annotation if any,
+//   - Machines failing to come up first because
+//     there is a chance that they are not hosting any workloads (minimize disruption).
+func sortMachinesToRemediate(machines []*clusterv1.Machine) {
+	sort.SliceStable(machines, func(i, j int) bool {
+		if annotations.HasRemediateMachine(machines[i]) && !annotations.HasRemediateMachine(machines[j]) {
+			return true
+		}
+		if !annotations.HasRemediateMachine(machines[i]) && annotations.HasRemediateMachine(machines[j]) {
+			return false
+		}
+		return machines[i].CreationTimestamp.After(machines[j].CreationTimestamp.Time)
```

Comment:
Should we do this similar to kcp and also handle the equal case (which propably never happens, but more likely to happen here than in KCP?

```suggestion
		// Use oldest (and Name) as a tie-breaker criteria.
		if machines[i].CreationTimestamp.Equal(machines[j].CreationTimestamp) {
			return machines[i].Name < machines[j].Name
		}
		return machines[i].CreationTimestamp.After(machines[j].CreationTimestamp)
```

Reply:
Or is it inetnional here and we want some "randomness"

Reply:
I dont think so, Updated as suggested to keep in sync with other places. Thanks